errbot/flow.py

```diff
@@ -122,7 +122,7 @@ class Flow(object):
     This is a live Flow. It keeps context of the conversation (requestor and context).
     Context is just a python dictionary representing the state of the conversation.
     """
-    def __init__(self, root: FlowRoot, requestor: Identifier, initial_context: Mapping[str, Any]):
+    def __init__(self, root: FlowRoot, requestor: Identifier, initial_context: Mapping[str, Any], dismiss_prompting=False):
```

Comment:
The naming `dismiss_prompting` makes us talk in double negatives (see below, `if not flow.dismiss_prompting`). I think it will read cleaner if we instead name it `next_step_hinting` (defaulting to `True`) so you can initialize a Flow with `Flow(..., next_step_hinting=False)` when you want to disable it.

In my experience this reads better and is easier to reason about.